Source/MSBuild.Sdk.Extras/Build/LanguageTargets/Desktop.targets

```diff
@@ -4,20 +4,27 @@
     <MSBuildAllProjects>$(MSBuildAllProjects);$(MSBuildThisFileFullPath)</MSBuildAllProjects>
   </PropertyGroup>
 
-  <!--
-    Only MSBuild Framework build includes Microsoft.Xaml.targets which redirects to MSBuildFrameworkToolsPath.
-    So to have Xaml Support, we need to use those targets instead of .NETCoreSdk targets
-  -->
-  <PropertyGroup Condition="$(_SdkDesktopTargets) == ''">
-    <_SdkDesktopTargets Condition="'$(_LanguageSourceName)' != 'FSharp'">$(MSBuildExtensionsPath)\$(_SdkVisualStudioVersion)\Bin\Microsoft.$(_LanguageSourceName).targets</_SdkDesktopTargets>
-    <_SdkDesktopTargets Condition="'$(_LanguageSourceName)' == 'FSharp'">$(MSBuildExtensionsPath)\Microsoft\VisualStudio\v$(_SdkVisualStudioVersion)\FSharp\Microsoft.FSharp.Targets</_SdkDesktopTargets>
+  <PropertyGroup Condition="'$(MicrosoftNETSdkTargetsPath)' == ''">
+    <MicrosoftNETSdkTargetsPath>$(ToolDepsJsonGeneratorProject.TrimEnd('GenerateDeps\GenerateDeps.proj'))</MicrosoftNETSdkTargetsPath>
+    <MicrosoftNETSdkTargetsPath Condition="!Exists('$(MicrosoftNETSdkTargetsPath)')">$(_NugetFallbackFolder.TrimEnd('..\..\..\..\NuGetFallbackFolder'))</MicrosoftNETSdkTargetsPath>
+    <MicrosoftNETSdkTargetsPath Condition="!Exists('$(MicrosoftNETSdkTargetsPath)')">$(NETCoreSdkBundledVersionsProps.TrimEnd('..\..\..\Microsoft.NETCoreSdk.BundledVersions.props'))</MicrosoftNETSdkTargetsPath>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(_FSharpTargetsPath)' == ''">
+    <_FSharpTargetsPath>$(MicrosoftNETSdkTargetsPath)Microsoft.NET.Sdk.FSharpTargetsShim.targets</_FSharpTargetsPath>
+    <_FSharpTargetsPath Condition="!Exists('$(_FSharpTargetsPath)')">$(MicrosoftNETSdkTargetsPath)Microsoft.NET.Sdk.FSharp.targets</_FSharpTargetsPath>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <_SdkLanguageTargets Condition="'$(_LanguageSourceName)' != 'FSharp'">$(MSBuildToolsPath)\Microsoft.$(_LanguageSourceName).targets</_SdkLanguageTargets>
```

Comment:
Why are we looking for the F# Targets this way instead of the VS directory?

Reply:
Why are these being set to a private variable and imported below instead of using `LanguageTargets`? 

https://github.com/onovotny/MSBuildSdkExtras/blob/rel/1.5/MSBuild.Sdk.Extras/build/platforms/languageTargets/Wpf.targets#L9-L10

Reply:
Still not following... I can see that you may want to add a second import, but I don't think we should be overriding the base language targets if they're the same? At least use the `LanguageTargets` value that's provided by default? 

I know there was one specific issue around WPF builds that needed to use the VS's version, but that was already handled? So I'm not really sure why stuff is being searched for via `MicrosoftNETSdkTargetsPath`?

Reply:
You were talking about this?

> Why are we looking for the F# Targets this way instead of the VS directory?

I wanted to make sure F# would have the same behaviour when under .NET SDK in this SDK too!

Reply:
We were already doing that here:

https://github.com/onovotny/MSBuildSdkExtras/blob/rel/1.5/MSBuild.Sdk.Extras/build/platforms/languageTargets/Wpf.targets#L8-L10

So why do we need the more complex logic here now?

Reply:
No, we're not doing that there! We're not importing from `Microsoft.NET.Sdk`